lib/ansible/modules/cloud/amazon/batch_compute_environment.py

```diff
@@ -0,0 +1,496 @@
+    required: true
+
+  compute_resource_type:
+    description:
+      - The type of compute resource.
+    required: true
+    choices: ["EC2", "SPOT"]
+
+  minv_cpus:
+    description:
+      - The minimum number of EC2 vCPUs that an environment should maintain.
+    required: true
+
+  maxv_cpus:
+    description:
+      - The maximum number of EC2 vCPUs that an environment can reach.
+    required: true
+
+  desiredv_cpus:
+    description:
+      - The desired number of EC2 vCPUS in the compute environment.
+
+  instance_types:
+    description:
+      - The instance types that may be launched.
+    required: true
+
+  image_id:
+    description:
+      - The Amazon Machine Image (AMI) ID used for instances launched in the compute environment.
+
+  subnets:
+    description:
+      - The VPC subnets into which the compute resources are launched.
+    required: true
+
+  security_group_ids:
+    description:
+      - The EC2 security groups that are associated with instances launched in the compute environment.
+    required: true
+
+  ec2_key_pair:
+    description:
+      - The EC2 key pair that is used for instances launched in the compute environment.
+
+  instance_role:
+    description:
+      - The Amazon ECS instance role applied to Amazon EC2 instances in a compute environment.
+    required: true
+
+  tags:
+    description:
+      - Key-value pair tags to be applied to resources that are launched in the compute environment.
+
+  bid_percentage:
+    description:
+      - The minimum percentage that a Spot Instance price must be when compared with the On-Demand price for that
+        instance type before instances are launched. For example, if your bid percentage is 20%, then the Spot price
+        must be below 20% of the current On-Demand price for that EC2 instance.
+
+  spot_iam_fleet_role:
+    description:
+      - The Amazon Resource Name (ARN) of the Amazon EC2 Spot Fleet IAM role applied to a SPOT compute environment.
+
+
+requirements:
+    - boto3
+extends_documentation_fragment:
+    - aws
+'''
+
+EXAMPLES = '''
+---
+- hosts: localhost
+  gather_facts: no
+  vars:
+    state: present
+  tasks:
+  - name: My Batch Compute Environment
+    batch_compute_environment:
+      compute_environment_name: computeEnvironmentName
+      state: present
+      region: us-east-1
+      compute_environment_state: ENABLED
+      type: MANAGED
+      compute_resource_type: EC2
+      minv_cpus: 0
+      maxv_cpus: 2
+      desiredv_cpus: 1
+      instance_types:
+        - optimal
+      subnets:
+        - my-subnet1
+        - my-subnet2
+      security_group_ids:
+        - my-sg1
+        - my-sg2
+      instance_role: arn:aws:iam::<account>:instance-profile/<role>
+      tags:
+        tag1: value1
+        tag2: value2
+      service_role: arn:aws:iam::<account>:role/service-role/<role>
+
+  - name: show results
+    debug: var=batch_compute_environment_action
+'''
+
+RETURN = '''
+---
+batch_compute_environment_action:
```

Comment:
currently returning changed, ansible_facts, and response

You can document the details like:
```
ansible_facts:
  description: helpful info
  type: complex
  returned: always
  contains:
    batch_compute_environment_action:
      description: describes what action was taken
    etc.
```

But I feel like what action was taken should be added to the returned response data (and not nest that data in response at all, since to access it the user will need to do something like registered_var.response.computeResources.tags[0] to get the associated tag) and ansible_facts should be removed. I don't know. Getting something called facts back from a module that isn't a _facts.py module doesn't seem intuitive. And the less nested the returned data is the easier to use.

Reply:
Removed ansible_facts. action taken is now returned in response data dictionary with the key batch_compute_environment_action and, is not nested in ansible_facts. See the sample in return doc string.